core/src/Streamly/Internal/Data/MutArray/Generic.hs

```diff
@@ -421,12 +426,8 @@ reallocWith label sizer reqSize arr = do
 --
 -- /Internal/
 {-# INLINE unsafeSnoc #-}
-unsafeSnoc, snocUnsafe :: MonadIO m => MutArray a -> a -> m (MutArray a)
-unsafeSnoc arr@MutArray {..} a = do
-    assert (arrStart + arrLen < arrTrueLen) (return ())
-    let arr1 = arr {arrLen = arrLen + 1}
-    unsafePutIndex arrLen arr1 a
-    return arr1
+snocUnsafe, unsafeSnoc :: MonadIO m => MutArray a -> a -> m (MutArray a)
+unsafeSnoc arr = snocNewEnd (arrEnd arr + 1) arr
```

Comment:
would be better to expand snocNewEnd code right here instead of using that function.